pkg/volume/flexvolume/driver-call.go

```diff
@@ -235,3 +241,29 @@ func handleCmdResponse(cmd string, output []byte) (*DriverStatus, error) {
 
 	return &status, nil
 }
+
+// getDriverCapabilities returns the reported capabilities as returned by driver's init() function
+func getDriverCapabilities(plugin *flexVolumePlugin) (*driverCapabilities, error) {
+	call := plugin.NewDriverCall(initCmd)
```

Comment:
I'm starting to lean towards separating out the driver capabilities introspection.

The problem is the new 'attach' code tosses in a monkey wrench, where to ascertain if the driver is attachable or not must be divined from the init() call.  Where it goes sideways is that init() (currently) must be called twice for the following flow:
```
ProbeVolumePlugins() -> NewFlexVolumePlugin() :: return attach ? flexVolumeAttachablePlugin : flexplugin
```
for which Init() is a method call on type _flexVolumePlugin_

Both call down into call.Run() via initCmd

If I work within how it is designed now, I would have to make the init() call in NewFlexVolumePlugin(), stuff away the result and then act on it in the Init() method.

If there is a consensus on splitting out the capabilities() handling as an independent function, I'd be happy to implement that in a future version (hopefully with backwards compatibility + deprecation msg).  For now, I will work within existing confines and stuff away the result in NewFlexVolumePlugin() and stuff away for Init() later.